blockindex/liquidstaking_bucket.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package blockindex
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/iotexproject/iotex-core/action/protocol/staking"
+	"github.com/iotexproject/iotex-core/blockindex/indexpb"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+)
+
+type (
+	// BucketInfo is the bucket information
+	BucketInfo struct {
+		TypeIndex  uint64
+		CreatedAt  time.Time
+		UnlockedAt time.Time
+		UnstakedAt time.Time
+		Delegate   string // owner address of the delegate
+		Owner      string
+	}
+
+	// BucketType is the bucket type
+	BucketType struct {
+		Amount      *big.Int
+		Duration    time.Duration
+		ActivatedAt time.Time
+	}
+
+	// Bucket is the bucket information including bucket type and bucket info
+	Bucket = staking.VoteBucket
+)
+
+func (bt *BucketType) toProto() *indexpb.BucketType {
```

Comment:
no public funcs for a public object?

Reply:
The methods may not be used outside this package, and the fields of the structure may be utilized instead